prow/plugins/blunderbuss/blunderbuss.go

```diff
@@ -271,34 +277,89 @@ func getReviewers(rc reviewersClient, author string, files []github.PullRequestC
 			continue
 		}
 		leafReviewers = leafReviewers.Union(fileUnusedLeafs)
-		reviewers.Insert(popRandom(fileUnusedLeafs))
+		if r := findReviewer(ghc, log, useStatusAvailability, &busyReviewers, &fileUnusedLeafs); r != "" {
+			reviewers.Insert(r)
+		}
 	}
 	// now ensure that we request review from at least minReviewers reviewers. Favor leaf reviewers.
 	unusedLeafs := leafReviewers.Difference(reviewers)
 	for reviewers.Len() < minReviewers && unusedLeafs.Len() > 0 {
-		reviewers.Insert(popRandom(unusedLeafs))
+		if r := findReviewer(ghc, log, useStatusAvailability, &busyReviewers, &unusedLeafs); r != "" {
+			reviewers.Insert(r)
+		}
 	}
 	for _, file := range files {
 		if reviewers.Len() >= minReviewers {
 			break
 		}
 		fileReviewers := rc.Reviewers(file.Filename).Difference(authorSet)
 		for reviewers.Len() < minReviewers && fileReviewers.Len() > 0 {
-			reviewers.Insert(popRandom(fileReviewers))
+			if r := findReviewer(ghc, log, useStatusAvailability, &busyReviewers, &fileReviewers); r != "" {
+				reviewers.Insert(r)
+			}
 		}
 	}
 	return reviewers.List(), requiredReviewers.List(), nil
 }
 
 // popRandom randomly selects an element of 'set' and pops it.
-func popRandom(set sets.String) string {
+func popRandom(set *sets.String) string {
 	list := set.List()
 	sort.Strings(list)
 	sel := list[rand.Intn(len(list))]
 	set.Delete(sel)
 	return sel
 }
 
+// findReviewer finds a reviewer from a set, potentially using status
+// availability.
+func findReviewer(ghc githubClient, log *logrus.Entry, useStatusAvailability bool, busyReviewers, targetSet *sets.String) string {
+	if useStatusAvailability {
```

Comment:
There'll be less indentation if you flip this around

Reply:
Yup.. I just saw this when I looked it over again. Fixed.